usrp_drivers/n200/n200_driver.cpp

```diff
@@ -507,32 +496,32 @@ void receive(zmq::context_t &driver_c, USRP &usrp_d, const DriverOptions &driver
   auto receive_channels = driver_options.get_receive_channels();
   stream_args.channels = receive_channels;
 
-  for (auto &ch : receive_channels) {
-    std::cout << "ch " << ch << std::endl;
-  }
   usrp_d.set_rx_rate(rx_rate_hz, receive_channels);  // ~450us
   uhd::rx_streamer::sptr rx_stream = usrp_d.get_usrp_rx_stream(stream_args);  // ~44ms
 
   /* 100 is the arbitrary scaling for the usrp_buffer_size
      so there won't be fragmentation and the while(1) loop below
      with the recv runs less times
   */
-  auto usrp_buffer_size = 50 * rx_stream->get_max_num_samps();
-  std::cout << "usrp_buffer_size " << usrp_buffer_size << std::endl;
-  // TODO: Put ringbuffer_size into the config file
+  auto usrp_buffer_size = 100 * rx_stream->get_max_num_samps();
   /* The ringbuffer_size is calculated this way because it's first truncated (size_t)
      then rescaled by usrp_buffer_size */
-  size_t ringbuffer_size = (size_t(500.0e6)/sizeof(std::complex<float>)/usrp_buffer_size) *
-                            usrp_buffer_size;
+  size_t ringbuffer_size = size_t(driver_options.get_ringbuffer_size()/
+                            sizeof(std::complex<float>)/
+                            usrp_buffer_size) * usrp_buffer_size;
+
+  SharedMemoryHandler shrmem(driver_options.get_ringbuffer_name());
+
+  shrmem.create_shr_mem(receive_channels.size() * ringbuffer_size * sizeof(std::complex<float>));
 
-  buffer.resize(receive_channels.size() * ringbuffer_size);
 
   std::vector<std::complex<float>*> buffer_ptrs_start;
 
   for(uint32_t i=0; i<receive_channels.size(); i++){
-    auto ptr = static_cast<std::complex<float>*>(buffer.data() + (i * ringbuffer_size));
+    auto ptr = static_cast<std::complex<float>*>(shrmem.get_shrmem_addr()) + (i * ringbuffer_size);
```

Comment:
Do you need to multiply i*ringbuffer_size in the calculation of  this ptr by sizeof(std::complex < float > ) ?

Reply:
No, I think automatic pointer arithmetic takes place here.